packages/mui-utils/src/deepmerge/deepmerge.ts

```diff
@@ -21,7 +22,7 @@ export interface DeepmergeOptions {
 }
 
 function deepClone<T>(source: T): T | Record<keyof any, unknown> {
-  if (React.isValidElement(source) || !isPlainObject(source)) {
+  if (React.isValidElement(source) || isValidElementType(source) || !isPlainObject(source)) {
```

Comment:
@siriwatknp why does `React.isValidElement(source)` not cover this case?

Reply:
`React.isValidElement` check React element like this `<Icon />` but it does not check for React component.

```js
createTheme({
  components: {
    MuiCheckbox: {
      defaultProps: {
        icon: <InfoSharp />, // React.isValidElement cover this
        component: CustomComponent, // React.isValidElement DOES not cover this
      },
    },
  },
});
```